x-pack/plugin/runtime-fields/src/main/java/org/elasticsearch/xpack/runtimefields/mapper/RuntimeScriptFieldMapper.java

```diff
@@ -69,8 +70,8 @@ protected void parseCreateField(ParseContext context) {
     }
 
     @Override
-    protected Object parseSourceValue(Object value, String format) {
-        throw new UnsupportedOperationException();
+    public ValueFetcher valueFetcher(SearchLookup lookup, String format) {
+        return docValuesFetcher(lookup, format);
     }
```

Comment:
After all that, this is the entire implementation! Everything else is shared and it all just works by virtue of supporting doc values.

Reply:
I didn't add any unit tests for this because we have unit tests for doc values. I did add a few integration tests just to make sure everything is plugged in.